source/guide_ejabberd.rst

```diff
@@ -0,0 +1,327 @@
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`supervisord <daemons-supervisord>`
+  * :manual:`domains <web-domains>`
+  * :manual:`firewall ports <basics-ports>`
+  * :manual:`web backends <web-backends>`
+
+This guide is based on the initial `pull request <https://github.com/Uberspace/lab/pull/444>`_ from `fm0de <https://github.com/fm0de>`_ and the `work from clerie <https://blog.clerie.de/ejabberd-auf-uberspace-installieren/>`_.
+
+Prerequisites
+=============
+
+Domain
+------
+
+Your ejabberd domain ``isabell.example`` needs to be setup. ejabberd then defaults to the subomains
+``upload.isabell.example``, ``conference.isabell.example`` , ``pubsub.isabell.example`` and ``proxy.isabell.example``
+::
+
+ [isabell@stardust ~]$ uberspace web domain add isabell.example
+ [isabell@stardust ~]$ uberspace web domain add conference.isabell.example
+ [isabell@stardust ~]$ uberspace web domain add upload.isabell.example
+ [isabell@stardust ~]$ uberspace web domain add pubsub.isabell.example
+ [isabell@stardust ~]$ uberspace web domain add proxy.isabell.example
+
+Also we need the SSL Certificates. Visit the domains in a browser or run:
+::
+
+ [isabell@stardust ~]$ curl https://isabell.example
+ [isabell@stardust ~]$ curl https://conference.isabell.example
+ [isabell@stardust ~]$ curl https://upload.isabell.example
+ [isabell@stardust ~]$ curl https://pubsub.isabell.example
+ [isabell@stardust ~]$ curl https://proxy.isabell.example
+
+
+Installation
+============
+
+Download, configure, compile and install ejabberd.
+Use the following options for ``./configure``:
+
+  * ``--prefix=$HOME/opt/ejabberd/``: Install to your personal uberspace
+  * ``--enable-user=$USER``: Allow execution of ejabberd as $USER
+  * ``--enable-mysql --enable-new-sql-schema``: optionally compile with mysql support
+  * ``--enable-sqlite``: optionally compile with sqlite support
+
+Run ``./configure --help`` to see all options.
+
+::
+
+ [isabell@stardust ~]$ cd $HOME/src/
+ [isabell@stardust src]$ wget https://github.com/processone/ejabberd/archive/20.03.tar.gz
+ [isabell@stardust src]$ tar xf 20.03.tar.gz
+ [isabell@stardust src]$ cd ejabberd-20.03/
+ [isabell@stardust ejabberd-20.03]$ ./autogen.sh
+ [isabell@stardust ejabberd-20.03]$ ./configure --enable-user=$USER --prefix=$HOME/opt/ejabberd --enable-mysql --enable-new-sql-schema
+ [isabell@stardust ejabberd-20.03]$ make install
+
+Make the control script available through ``.bash_profile``:
+
+.. code-block:: bash
+
+ # Add ejabberd binaries to PATH
+ export PATH=$HOME/opt/ejabberd/sbin:$PATH
```

Comment:
since there is just `ejabberdctl` in that directory, a `ln -s ~/opt/ejabberd/sbin/ejabberdctl ~/bin/` seems like an easier solution. What do you think?

Reply:
also fine with me.